File: regexlib/2021-5-15/python_re2_test_file/regexlib_228.py
# 228
# ^([A-Za-z]|[A-Za-z][0-9]*|[0-9]*[A-Za-z])+$
# EXPONENT
# nums:5
# EXPONENT AttackString:""+"AA"*8+"!1 __EOD(ii2)"

import re2 as re
from time import perf_counter

regex = """^([A-Za-z]|[A-Za-z][0-9]*|[0-9]*[A-Za-z])+$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "AA" * i * 1 + "!1 __EOD(ii2)"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")

File: regexlib/javascript_test_file/regexlib_228.js
/* 228
 * ^([A-Za-z]|[A-Za-z][0-9]*|[0-9]*[A-Za-z])+$
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:""+"AA"*8+"!1 __EOD(ii2)"
 */
var REGEX = /^([A-Za-z]|[A-Za-z][0-9]*|[0-9]*[A-Za-z])+$/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + 'AA'.repeat(i*1) + '!1 __EOD(ii2)'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}